### PR TITLE
Fix mwa_corr_fits start_flag

### PR DESCRIPTION
**Pull request by PyxieLouStar**
Update deriving `start_flag` from the 'GOODTIME' metafits field so that `start_flag` is an integer multiple of the integration time.
## Description

If `start_flag` is not an integer multiple of the integration time, round `start_flag` up to the nearest multiple of the integration time.
## Motivation and Context
Subtracting the 'GOODTIME' field from the first integration time can give a `start_flag` that is not an integer multiple of the integration time, causing the read to fail.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [x] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

### CODE REVIEW

**Comment by PyxieLouStar** [comment]
This bug was causing an error in the read of 2014 data files taken at 2.0 seconds integration time. The fix has been tested on those files.

File: pyuvdata/uvdata/tests/test_mwa_corr_fits.py [hunk 1]
```diff
@@ -788,6 +788,22 @@ def test_start_flag_bad_string():
...
+    """Test goodtime returning a start_flag smaller than integration time."""
+    uv = UVData()
+    new_meta = str(tmp_path / "1131733552_goodtime.metafits")
+    with fits.open(filelist[0]) as meta:
+        meta[0].header["GOODTIME"] = 1447698337.25
+        meta.writeto(new_meta)
+    uv.read(
+        [new_meta, filelist[1]], flag_init=True, start_flag="goodtime",
+    )
+    # goodtime > start_time so all data should be flagged
```
Comment (bhazelton) [inline pyuvdata/uvdata/tests/test_mwa_corr_fits.py:803]: I'm confused about this comment. I would think that if goodtime is greater than start_time then some data should be flagged, but I don't see why it means that all data should be flagged.

Comment (PyxieLouStar) [reply pyuvdata/uvdata/tests/test_mwa_corr_fits.py:803]: There is only one time integration in the test file, so all the data should be flagged. I could change the test to have more time integrations; that would be more robust.

Comment (bhazelton) [reply pyuvdata/uvdata/tests/test_mwa_corr_fits.py:803]: I think the test is fine, can you just clarify the comment a little?

Comment (PyxieLouStar) [reply pyuvdata/uvdata/tests/test_mwa_corr_fits.py]: Done!

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 - A bug in reading in the MWA beam that caused the beam to be rotated by 90 degrees.
+- A bug in mwa_corr_fits that didn't round start_flag to the nearest multiple of
```
Comment (bhazelton) [inline CHANGELOG.md:11]: This should be at the top (reverse chronological order).

Comment (PyxieLouStar) [reply CHANGELOG.md:10]: Right, sorry about that. It's fixed now.

**Review by bhazelton — APPROVED** [review]
Looks good, thanks!

**Pull request merged**